### PR TITLE
Add Device: Innovaphone

### LINKED ISSUE #9532: New Device: Innovaphone IP6010
**Please do not post bugs or ask questions**

**For help and support you can use our [Discord server](https://t.libren.ms/discord) or our [community site](https://community.librenms.org).**

Please confirm each of the sections below by putting an `x` in the box like `[x]`.

- [x] Please provide **ALL** info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).
- [x] Please provide as much detail as possible.
- [x] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.

Love librenms? Please consider supporting our collective:
👉  https://t.libren.ms/donations

Brand: Innovaphone
Type: IP6010
Logo: https://kreuze.nl/wp-content/uploads/2016/10/innovaphone_Logo_RGB_Claim_2502x750.jpg

Discovery: https://p.libren.ms/view/30dcfd97
Poller: https://p.libren.ms/view/3a7f8c6a
SnmpWalk: https://p.libren.ms/view/635f473f
inno-mib-v7.txt: https://p.libren.ms/view/5697dec0

### PR DESCRIPTION
**Pull request by mpikzink**
fixes #9532
Add Innovaphone PBX telephone system

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: mibs/INNO-MIB [hunk 1]
```diff
@@ -0,0 +1,576 @@
+INNO-MIB   DEFINITIONS ::= BEGIN
```
Comment (PipoCanaja) [inline mibs/INNO-MIB]: Hi
Could you please create and move this mib file into its directory ?
Thanx

**Comment by PipoCanaja** [comment]
Hi
Please have a look at travis results, and correct the test data: 
```
-            'state_name' => 'l3Table'
+            'state_name' => 'l3Label'
```

File: includes/definitions/discovery/innovaphone.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+mib: INNO-MIB
+modules:
+    sensors:
+        state:
+            data:
+                -
```
Comment (PipoCanaja) [inline includes/definitions/discovery/innovaphone.yaml]: This YAML state definition looks wrong . You walk a table but do not use the index you get and instead hardcode the index. That cannot give the proper result (at least not every time). Like below, you hardcode PRI1-L2 as the index, and may display the wrong value for another index.

File: includes/definitions/discovery/innovaphone.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                    state_name:  l3Label
+                    states:
+                        - { descr: none, graph: 0, value: 1, generic: 3 }
+                        - { descr: other,   graph: 0, value: 2, generic: 3 }
+                        - { descr: etsi,   graph: 0, value: 3, generic: 0 }
+                        - { descr: qsig,   graph: 0, value: 23, generic: 0 }
+        count:
+            data:
+                -
+                    oid: l3Table
```
Comment (PipoCanaja) [inline includes/definitions/discovery/innovaphone.yaml]: same here. You hardcode PRI1 in the num_oid even if the device could use another interface name (and then discovery will run but polling will fail)

File: includes/discovery/sensors/state/innovaphone.inc.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    $num_oid = '.1.3.6.1.4.1.6666.2.1.1.1.4.';
+    foreach ($oids as $index => $entry) {
+        $name='Voice If ' . hex2bin(strtr($entry['voiceIfName'], [' '=>'']));
+        //Discover Sensors
+        discover_sensor($valid['sensor'], 'state', $device, $num_oid.$index, $index, $state_name, $name, '1', '1', null, null, null, null, $entry['voiceIfState'], 'snmp', $index);
+
+        //Create Sensor To State Index
+        create_sensor_to_state_index($device, $state_name, $index);
+    }
+}
```
Comment (PipoCanaja) [inline includes/discovery/sensors/state/innovaphone.inc.php]: This code looks more promising.

File: includes/definitions/discovery/innovaphone.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                    descr: 'Interface Mode'
+                    state_name: L2Mode
+                    states:
+                        - { descr: TE, graph: 0, value: 1, generic: 0 }
+                        - { descr: NT, graph: 0, value: 2, generic: 0 }
+
+                -
+                    oid: l1BriTable
+                    value: l1BriState
+                    num_oid: '.1.3.6.1.4.1.6666.1.1.1.3'
```
Comment (PipoCanaja) [inline includes/definitions/discovery/innovaphone.yaml]: This num OID is the same as the previous one (so one of the 2 is wrong) and does not include index.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-62-release-changelog-march-2020/11423/1

**Pull request merged**